.circleci/config.yml

```diff
@@ -1,241 +1,216 @@
 
-  defaults: &defaults
+# -------------------------
+#   EXECUTORS
+# -------------------------
+executors:
+  workbench-executor:
+    environment:
+      <<: *default-environment
     docker:
-      - image: << pipeline.parameters.workbench_image >>
+      - image: << pipeline.parameters.workbench-image >>
     working_directory: ~/workbench
 
-  java_defaults: &java_defaults
-    <<: *defaults
+  db-executor:
+    docker:
+      - image: << pipeline.parameters.workbench-image >>
+      - image: << pipeline.parameters.db-image >>
+        environment:
+          <<: *db-environment
     environment:
-      <<: *java_env
-
-  # See https://circleci.com/docs/2.0/workflows/#git-tag-job-execution
-  # Job runs for master branch only
-  filter_only_master_branch: &filter_only_master_branch
-    tags:
-      ignore: /.*/
-    branches:
-      only: master
+      <<: *default-environment
+      <<: *db-environment
+    working_directory: ~/workbench
 
-  # Job runs for no branch and only for tags starting with ‘v’
-  filter_only_release_tags: &filter_only_release_tags
-    branches:
-      ignore: /.*/
-    tags:
-      # regex explanation:
-      # "^" asserts position at start of a line.
-      # "v" matches the character "v" literally (case sensitive).
-      # "[0-9]+" match a single numerical digit (the + quantifier, matches between one and unlimited times, as many times as possible).
-      # "-" matches the character "-" literally.
-      # "rc" matches the character "rc" literally (case sensitive).
-      # "$" asserts position at the end of a line.
-      # Workbench release tag format example: v5-3-rc1
-      only: /^v[0-9]+-[0-9]+-rc[0-9]+$/
-
-  # Job runs for PR branches only
-  filter_only_pr_branch: &filter_only_pr_branch
-    tags:
-      ignore: /.*/
-    branches:
-      ignore: master
+  puppeteer-executor:
+    docker:
+      - image: << pipeline.parameters.workbench-image >>
+    working_directory: ~/workbench
+    environment:
+      <<: *puppeteer-environment
 
 # -------------------------
 #   COMMANDS
 # Refers to https://circleci.com/docs/2.0/reusing-config/#authoring-reusable-commands
 # -------------------------
 commands:
-  activate_service_account_credential:
+  gcloud-auth-login:
     description: "Activiate CircleCI service account credential"
     steps:
       - run:
-          name: Activate CircleCI service account credential
           working_directory: ~/workbench
-          # Used to call gsutil from the circle environment.
           command: ci/activate_creds.sh api/circle-sa-key.json
 
-  checkout_init_git:
+  checkout-code:
     description: "git checkout and update submodules"
     steps:
       - checkout
       - run:
-          name: Update git submodules
           command: git submodule update --init --recursive
 
-  halt_if_code_unchanged:
-    description: "Halt job if no code changed"
+  store-commit-message:
+    description: "Export Git commit message to environment variable"
```

Comment:
aside: I wonder why we have to do this. Why wouldn't CircleCI have this built in? Or is there an off-the-shelf PR trigger orb that simplifies this stuff?